### ISSUE TITLE
Pi 4 PXE with kickstart install seems to be unbootable

### ISSUE THREAD

**Issue by jontywigster** [vmware/photon#1377]
### Describe the bug

A PXE install of Photon OS 4 with a kickstart file seems to result in an unbootable system. After the installation finishes, the machine reboots and the following appears - 

<img width="206" alt="image" src="https://user-images.githubusercontent.com/53944826/200680680-17dfaadb-d48f-4c6d-b14a-44dbb7765369.png">

A PXE install of Photon OS 4 without specifying a kickstart files boots normally.

### Reproduction steps

1. For a Raspberry Pi 4 with the latest UEFI firmware from [(https://github.com/pftf/RPi4)], Create a grub2 menu for PXE and specify, e.g., 

`linux (http)/a.b.c.d/photon/isolinux/vmlinuz root=/dev/ram0 ramdisk_size=1500000 loglevel=3 repo=http://a.b.c.d/photon/RPMS/ insecure_installation=1 ks=http://a.b.c.d/kickstart_file`

NB - the specified kickstart file shall be valid

2. Compare the result of above on the same PI and grub2 PXE environment, with the ks parameter excluded -

`linux (http)/a.b.c.d/photon/isolinux/vmlinuz root=/dev/ram0 ramdisk_size=1500000 loglevel=3 repo=http://a.b.c.d/photon/RPMS/ insecure_installation=1`

### Expected behavior

The kickstart-specified file would result in a bootable system

### Additional context

I'm new to Photon OS so probably don't know what I'm doing. I happened to try x64 first with, basically, the same configuration. My x64 setup works. On my Pi, I get the same failure behaviour whether PXE is installing directly to the PI, or the Pi is running VMWare ESXi 7 and the PXE install is to a VM within ESXi. If I don't specify a kickstart file, the install works. Because kickstart works on x64 for me, I'm fairly sure I'm not making a mistake in the kickstart file. 

On the Pi, towards the end of a failing install I see - 
 
<img width="645" alt="image" src="https://user-images.githubusercontent.com/53944826/200686729-f60c367d-1c15-4ddc-836a-e5c75f53b35a.png">

I don't know if those errors are relevant. In any case, I saw https://github.com/vmware/photon/issues/1089 but adding 'glibc-i18n' to the kickstart 'additional_packages' parameter, or in a postinstall parameter script, results in the same error messages.

**Comment by dcasota** [comment]
Hi, 
some thoughts
- is in vm settings uefi secure boot disabled as well and then pxe-install Photon OS 4 ?
- [minimal.spec](https://github.com/vmware/photon/blob/4.0/SPECS/minimal/minimal.spec) in 4.0 does not contain e.g. glibc-i18n. It may be necessary first from modified minimal.spec to create a (custom bootable) home iso/ova.
- unfortunately there is no dynamic web-based media creator with selectable options e.g. x64/aarch64, with/without secure boot, with/without kickstart, Photon OS flavor, default language,, etc.

**Comment by jontywigster** [comment]
Grüezi,

Thanks to your thoughts, I realised I missed something obvious and embarrassing...

The ks parameter points at a web server, and this server returns a slightly edited version of a kickstart file that is used as a template, e.g., the hostname is set. There was a bug in my code that meant the web server was not using my template. In other words, I added glibc-i18n to my kickstart template but the actual file sent to the installer would not have included it. Oops. A PXE install now works as expected on my poor Pi.

Sorry!
